src/main/api/fileStore.ts

```diff
@@ -0,0 +1,79 @@
+import path from 'path'
+
+import { app, ipcRenderer, ipcMain } from 'electron'
+import * as fs from 'fs-extra'
+
+import { ApiFileStoreService, StoreFileData, StoreFileName } from '../../shared/api/types'
+import { getStoreFilesIPCMessages } from '../../shared/ipc/fileStore'
+
+const APP_NAME = app?.name ?? 'ASGARDEX'
+
+const APP_DATA_DIR = path.join(app?.getPath('appData') ?? './testdata', APP_NAME)
+export const STORAGE_DIR = path.join(APP_DATA_DIR, 'storage')
+
+// If `fullFilePathname' does not exist, `fs.remove` silently does nothing.
+// @see https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md
+const removeFile = async (fullFilePathname: string) => fs.remove(fullFilePathname)
+
+const isFileExist = async (fullFilePathname: string) => fs.pathExists(fullFilePathname)
+
+const getFileContent = async <T>(fullFilePathname: string, defaultValue: T): Promise<T> => {
+  // If wile does not exist create a new one with defaultValue as content
+  if (!(await isFileExist(fullFilePathname))) {
+    await fs.writeJSON(fullFilePathname, { ...defaultValue })
+    return Promise.resolve(defaultValue)
+  }
+
+  try {
+    // Try to read and parse appropriate JSON store-file
+    const fileContent = await fs.readJSON(fullFilePathname)
+    // Combine file content with provided default value
+    return Promise.resolve({ ...defaultValue, ...fileContent })
+  } catch {
+    // Resolve with default value in case of any error
+    return Promise.resolve(defaultValue)
```

Comment:
What's the reason to never fail `getFileContent` by returning `defaultValue`? With it, `saveFile` will never fail as well, but in some cases we might want to know that we can't write something to disk, don't we?

Reply:
for `getFileContent` i believe it makes sense to have default values for real world app.

about `saveFile` there is [fs.writeJSON](https://github.com/thorchain/asgardex-electron/pull/795/files#diff-1a3af8eeb2e0ceea21cd19c157aab914739e4ad090353a6710c61ab20917a93bR43) which will fail in case when it can not write to the file and all resulted `Promise` will be rejected. do i miss anything ?

Reply:
[Current implementation of `getFileContent`](https://github.com/thorchain/asgardex-electron/blob/0d27dbd376f39c287293a955d9b733d0910570a4/src/main/api/fileStore.ts#L16-L32) will never fail, there is no `try/catch` or something [around fs.writeJSON](https://github.com/thorchain/asgardex-electron/blob/0d27dbd376f39c287293a955d9b733d0910570a4/src/main/api/fileStore.ts#L19). We can do it, just let you know about ...